superbench/benchmarks/micro_benchmarks/computation_communication_overlap.py

```diff
@@ -0,0 +1,242 @@
+        )
+        self._parser.add_argument(
+            '--q',
+            type=int,
+            default=1400,
+            required=False,
+            help='The Q dim of Tensor(P, Q) to transfer in Nccl AllReduce.',
+        )
+        self._parser.add_argument(
+            '--ratio',
+            type=int,
+            default=30,
+            required=False,
+            help='The execution ratio number between computation kernel and nccl kernel',
+        )
+        self._parser.add_argument(
+            '--kernel',
+            type=ComputationKernelType,
+            default=[ComputationKernelType.MUL],
+            required=False,
+            help='Computation kernel type. E.g. {}.'.format(' '.join(ComputationKernelType.get_values())),
+        )
+        self._parser.add_argument(
+            '--num_warmup',
+            type=int,
+            default=64,
+            required=False,
+            help='The number of warmup step.',
+        )
+        self._parser.add_argument(
+            '--num_steps',
+            type=int,
+            default=2048,
+            required=False,
+            help='The number of test step.',
+        )
+
+    def _preprocess(self):
+        """Preprocess/preparation operations before the benchmarking.
+
+        Return:
+            True if _preprocess() succeed.
+        """
+        if not super()._preprocess():
+            return False
+
+        try:
+            torch.distributed.init_process_group(backend='nccl')
+            self.__world_size = int(os.environ['WORLD_SIZE'])
+            self.__local_rank = int(os.environ['LOCAL_RANK'])
+        except BaseException as e:
+            self._result.set_return_code(ReturnCode.DISTRIBUTED_SETTING_INIT_FAILURE)
+            logger.error('Initialize distributed env failed - benchmark: {}, message: {}.'.format(self._name, str(e)))
+            return False
+
+        if torch.cuda.is_available():
+            torch.cuda.set_device(self.__local_rank)
+
+        return True
+
+    def __kernel_nccl_pipeline(self, kernel, matA, matB, stages, message, times):
+        """Computation and nccl kernel pipeline with single GPU.
+
+        Args:
+            kernel (ComputationKernelType): the type of the computation kernel to run
+            matA (list[tensor]): the matrix list used in matmul or mul for every stage
+            matB (tensor): the matrix used in matmul
+            stages (int): the ratio number of computation kernel and communication kernel
+            message(tensor): the data used to be transferred through NCCL
+            times(int): number of times in one step to run
+
+        Return:
+            True of False: if computation kernel type is invalid, return False, else, return True
+        """
+        for i in range(times):
+            if self.__world_size > 1:
+                torch.distributed.all_reduce(message, op=torch.distributed.ReduceOp.SUM, async_op=True)
```

Comment:
The ratio between NCCL and Computation is very critical for this testing. For example, in K80, the computation will be very slow. Do we need to keep different setting for different GPU? @guoshzhao .

Reply:
ratio is a argument, I think it's the right way to control the benchmark behavior for different GPUs.